tests/perf/perf_tests.cc

```diff
@@ -203,6 +204,119 @@ static inline std::ostream& operator<<(std::ostream& os, duration d)
     return os;
 }
 
+/**
+ * A column object encapsulates the logic needed to print one
+ * type of result value, usually as a column (or a json attribute).
+ *
+ * This allows all printers to share a common view of the available
+ * columns.
+ */
+struct column {
+    static constexpr int default_width = 11;
+
+    using print_fn = std::function<void(const column&, std::FILE *file, const result& r)>;
+
+    template <typename F>
+    column(sstring header, int prec, F fn) : header{header}, prec{prec} {
```

Comment:
nit, i think `std::string_view` is good enough here. because what we want is just a type which is convertible to `std::string`. and the callers of this constructor always pass a string literal for `header`. if you agree with my reasoning, we might need to 

- update the member variable of `header` to be `std::string_view` as well.
- s/c.header/std::string{c.header}/ in `json_printer::print_result()`